Improve editing of file prompts and update key descriptions

clemera opened the pull request.

okamsn commented:
Would you also mention that mouse bindings that were added, which work like `RET` and `TAB`?

Thank you.

raxod502 commented on README.md:
```diff
@@ -163,9 +163,13 @@ editing bindings. So, for example:
   deletion commands work as usual.
 * To delete your current input, just use `C-a C-k`.
-* To go up a directory, use `M-DEL` to delete it in the minibuffer.
+* To go up a directory, use `M-DEL` or `ESC C-backspace` (bound to
```
Why `ESC C-backspace` instead of `C-M-DEL`?

raxod502 replied:
Oh, because `C-M-DEL` doesn't do the same thing somehow. I cannot say I understand why, but fair enough :P

raxod502 replied:
I will point out, by the way, that `backward-kill-sexp` doesn't actually move you one level up necessarily, if there are spaces in the path name.

okamsn replied:
On Linux, `C-M-backspace` is sometimes set to kill the X server, which can be inconvenient. Using `ESC C-backspace` is therefore safer.

clemera replied:
> Oh, because C-M-DEL doesn't do the same thing somehow. I cannot say I understand why, but fair enough :P

I thought it isn't bound to that key by default because `(key-binding (kbd "C-M-DEL"))` returns `nil` for me. I have rebound it manually to that sequence. Maybe we should do that by default? If `C-M-DEL` kill the X server the user can rebind it.

clemera replied:
> I will point out, by the way, that backward-kill-sexp doesn't actually move you one level up necessarily, if there are spaces in the path name.

Thanks, we could fix that by modifying the syntax table when reading file names (adding space char as symbol constituents), would there be any downsides to this?

clemera commented:
> Would you also mention that mouse bindings that were added, which work like RET and TAB?

Done

clemera commented:
I have added the `C-M-DEL` binding and also changed the syntax table for reading file names to include spaces as symbol constituents.

clemera commented:
I have added the `C-M-DEL` binding and also changed the syntax table for sexp commands when reading file names to include spaces as symbol constituents.

raxod502 commented on selectrum.el:
```diff
@@ -1181,6 +1185,44 @@ minibuffer."
+
+(defun selectrum-backward-kill-sexp ()
+  "Forward to `backward-kill-sexp'.
+Adjusting sytnax table for file completions to
```
```suggestion
Adjusting syntax table for file completions to
```

raxod502 commented on selectrum.el:
```diff
@@ -1181,6 +1185,44 @@ minibuffer."
+
+(defun selectrum-forward-sexp ()
+  "Forward to `forward-sexp'.
+Adjusting sytnax table for file completions to
```
```suggestion
Adjusting syntax table for file completions to
```

raxod502 commented on selectrum.el:
```diff
@@ -1181,6 +1185,44 @@ minibuffer."
+
+(defun selectrum-backward-sexp ()
+  "Forward to `backward-sexp'.
+Adjusting sytnax table for file completions to
```
```suggestion
Adjusting syntax table for file completions to
```

clemera replied:
Thkans! Thkans! Thkans! ;)

raxod502 commented on README.md:
```diff
@@ -163,9 +165,11 @@ editing bindings. So, for example:
   deletion commands work as usual.
 * To delete your current input, just use `C-a C-k`.
-* To go up a directory, use `M-DEL` to delete it in the minibuffer.
+* To edit the input use `M-DEL` or `C-M-DEL` to go up a directory.
```
Maybe explain the difference between these two commands?

clemera replied:
Done

clemera commented:
> I suspect we could set the syntax table for the entire minibuffer temporarily and then it would work for all editing commands

I did that first by let binding `minibuffer-local-filename-syntax` and then thought about the bad effect for recursive sessions. But now I realize we could just set it within the current session by using the `(:append FUNCTION)` feature of `minibuffer-with-setup-hook`.

okamsn commented on README.md:
```diff
@@ -163,9 +165,12 @@ editing bindings. So, for example:
 * To delete your current input, just use `C-a C-k`.
-* To go up a directory, use `M-DEL` to delete it in the minibuffer.
+* To edit by word units use `M-DEL` like usual. To go up a directory
+  you can use `C-M-DEL` (bound to `backward-kill-sexp`).
```
Would you mind including something like the quote below? I know that some distributions are moving away from enabling this feature, and that the binding is easily rebound within Emacs, but I think it is good to be clear about what can happen. 

A user who accidentally kills their programs by trying this binding might report this as an issue, and I think it would be better to get ahead of such things.

> Be aware that on some Linux distributions, this binding is used to kill the X server, which can cause loss of unsaved work. In that case, `backward-kill-sexp` is also by default bound to `ESC C-DEL`.

clemera replied:
Good idea, thanks!

raxod502 commented:
Nice!
